src/usecases/link-car.ts

```diff
@@ -0,0 +1,39 @@
+import { newCar, Car, CarRepository } from "../entities/car";
+
+export interface LinkCarRequest {
+  id: String;
+  company?: String;
+  model?: Number;
+  make?: String;
+  color?: String;
+}
+
+export interface CarLinkingService {
+  linkCar: (req: LinkCarRequest) => Car;
+}
+
+export function newCarLinkingService(
+  carRepo: CarRepository
+): CarLinkingService {
+  return Object.freeze({
+    linkCar: makeLinkCar(carRepo),
+  });
+}
+
+function makeLinkCar(carRepo: CarRepository): (req: LinkCarRequest) => Car {
+  return function linkCar(req: LinkCarRequest): Car {
+    const car = linkCarRequestToNewCar(req);
+    carRepo.save(car);
+    return car;
+  };
+}
+
+function linkCarRequestToNewCar(req: LinkCarRequest): Car {
+  return newCar({
+    id: req.id,
+    company: req.company ? req.company : "",
+    model: req.model ? req.model : 0,
+    make: req.make ? req.make : "",
+    color: req.color ? req.color : "",
+  });
+}
```

Comment:
made a separate function for converting request object to car object

Reply:
I think we should consider making this idempotent. Makes it much easier to consume from a client/FE perspective.
With this implementation, the behavior would be such:
1. user sends req => { id: "ABC-123", company: "Toyota" }
db state => { id: "ABC-123", company: "Toyota" }
2. user sends req later => { id: "ABC-123", color: "white" }
db state => { id: "ABC-123", company: "", color: "white" }

not sure if that's ideal.